Add ACD installation on SmartProxyServer (#534)
theforeman/foreman-documentation: guides/common/modules/proc_acd_installation.adoc

[id="{context}_installation"]
= Installing ACD on {ProjectServer}

ACD has a {Project} plugin (`foreman_acd`) and a {SmartProxy} plugin (`smart_proxy_acd`).

ifdef::foreman-deb[]
Running ACD on Debian and Ubuntu is currently untested.
endif::[]

.Procedure
. Install ACD on your {ProjectServer}:
+
[options="nowrap", subs="verbatim,quotes,attributes"]
----
# {installer-scenario} --enable-foreman-plugin-acd --enable-foreman-proxy-plugin-acd
----

[id="{context}_installation_{SmartProxyServer}"]
= Installing ACD on {SmartProxyServer}

.Prerequisite
* {ProjectServer} with ACD xref:{context}_installation[installed].

.Procedure
. Install ACD on your {SmartProxyServer}:
+
[options="nowrap", subs="verbatim,quotes,attributes"]
----
# {installer-scenario-smartproxy} --enable-foreman-proxy-plugin-acd
----

> ifdef::foreman-el,foreman-deb,katello,satellite[]

Author: Removed.

[id="{context}_installation_{ProjectServer}"]
= Installing ACD on {ProjectServer}

> ACD consists of two plugins: `foreman_acd` and `smart_proxy_acd`.

Reviewer: Just a thought, but perhaps something like this?
```suggestion
ACD has a Foreman plugin (`foreman_acd`) and a Foreman Proxy plugin (`smart_proxy_acd`).
```

Author: Added!

endif::[]

.Procedure
. Install ACD on your {ProjectServer}:
+
[options="nowrap", subs="verbatim,quotes,attributes"]
----

> # {foreman-installer} --enable-foreman-plugin-acd --enable-foreman-proxy-plugin-acd

Reviewer: Perhaps this is better:
```suggestion
# {installer-scenario} --enable-foreman-plugin-acd --enable-foreman-proxy-plugin-acd
```

Then `This guide assumes you already have a working` becomes less relevant. You still need to have set up the repositories and installed the installer itself, there is no longer a need to have ran the installer before.

Author: I've changed/removed this. I think for a usage guide for a plugin, it's ok to assume users have a working Foreman installation.

----

[id="{context}_installation_{SmartProxyServer}"]
= Installing ACD on {SmartProxyServer}

.Prerequisite
* {ProjectServer} with ACD xref:{context}_installation[installed].

.Procedure
. Install ACD on your {SmartProxyServer}:
+
[options="nowrap", subs="verbatim,quotes,attributes"]
----

> # {foreman-installer} --enable-foreman-proxy-plugin-acd

Reviewer: Perhaps this is better:
```suggestion
# {installer-scenario-smartproxy} --enable-foreman-proxy-plugin-acd
```

That way it also works as a fresh installation rather than only on an existing one.

Author: Thanks, fixed!

# {foreman-maintain} service restart
----

[id="{context}_installation_{SmartProxyServer}"]
= Installing ACD on {SmartProxyServer}

> You need to install two packages to use ACD on a {SmartProxy}.

Reviewer 2: ```suggestion
You need to install two packages to use ACD on {SmartProxy}.
```

Reviewer 2: You could always use "your" to get around it if you think it sounds weird!

Author: Fixed.

.Procedure
ifdef::foreman-el,foreman-deb,katello,satellite[]

> . Create a new yum repository `/etc/yum.repos.d/foreman-plugins.repo` on your {SmartProxyServer}:

Reviewer 2: Create implies new, so no need to say new! 
```suggestion
. Create a yum repository `/etc/yum.repos.d/foreman-plugins.repo` on your {SmartProxyServer}:
```

Author: Fixed.

# {foreman-maintain} service restart
----

[id="{context}_installation_{SmartProxyServer}"]
= Installing ACD on {SmartProxyServer}

> You need to install two packages to use ACD on your {SmartProxy}.

Author: Removed.

.Prerequisites
* {ProjectServer} with ACD xref:{context}_installation[installed].

.Procedure
. Install the required packages on your {SmartProxyServer}:
+
[source,bash]
----
# yum install tfm-rubygem-smart_proxy_acd tfm-rubygem-smart_proxy_acd_core
----
. Restart the {SmartProxyServer} services:
+
[source,bash]
----
# systemctl restart foreman-proxy smart_proxy_dynflow_core
----
. Refresh {SmartProxy} features on {ProjectServer}:
+
In the {Project} web UI, navigate to *Infrastructure* > *{SmartProxies}* and select your {SmartProxyServer}.

> From the *Actions* list, click *Refresh*.

Author: Thanks, fixed.

# {foreman-maintain} service restart
----

> [id="{context}_installation_{SmartProxyServer}"]

Reviewer 2: @adahms would you know offhand if this heading styling will cause us problems long term?

= Installing ACD on {SmartProxyServer}

You need to install two packages to use ACD on your {SmartProxy}.

.Procedure
ifdef::foreman-el,foreman-deb,katello,satellite[]
. Create a yum repository `/etc/yum.repos.d/foreman-plugins.repo` on your {SmartProxyServer}:
+
[source,bash,subs="+quotes,attributes"]
----
[foreman-plugins]
name=Foreman plugins
baseurl=https://yum.theforeman.org/plugins/nightly/el7/x86_64/
enabled=1
gpgcheck=0
----
endif::[]
. Install the required packages on your {SmartProxyServer}:
+
[source,bash]
----
# yum install tfm-rubygem-smart_proxy_acd tfm-rubygem-smart_proxy_acd_core
----
. Restart the {SmartProxy} services:
+
[source,bash]
----
# systemctl restart foreman-proxy
----
. Refresh {SmartProxy} features on {ProjectServer}:
+

> Navigate to *Infrastructure* > *Smart Proxies* and select your {SmartProxyServer}.

Reviewer 2: ```suggestion
In the {Project} web UI, navigate to *Infrastructure* > *{SmartProxies}* and select your {SmartProxyServer}.
```

Reviewer 2: Should this be a step?

Author: Fixed.

I don't know how to shorten this further. IMHO "refresh this feature" is the step and the bottom two lines are the explanation on how to do this.

> Click the *Refresh* button in the *Actions* drop down menu.

Reviewer 2: ```suggestion
From the *Actions* list, click *Refresh*.
```
The most interesting part of the IBM style guide is "Graphical User Interface Elements" - I'm not sure if you have a copy yet, but it is great.

Author: Fixed.

And yes, I finally got my own copy.

# {foreman-maintain} service restart
----

[id="{context}_installation_{SmartProxyServer}"]
= Installing ACD on {SmartProxyServer}

You need to install two packages to use ACD on your {SmartProxy}.

.Prerequisites
* {ProjectServer} with ACD xref:{context}_installation[installed].

.Procedure
ifdef::foreman-el,foreman-deb,katello,satellite[]
. Create a yum repository `/etc/yum.repos.d/foreman-plugins.repo` on your {SmartProxyServer}:
+
[source,bash,subs="+quotes,attributes"]
----
[foreman-plugins]
name=Foreman plugins
baseurl=https://yum.theforeman.org/plugins/nightly/el7/x86_64/
enabled=1
gpgcheck=0
----

> endif::[]

Author: I've removed this first step.

. Install the required packages on your {SmartProxyServer}:
+
[source,bash]
----
# yum install tfm-rubygem-smart_proxy_acd tfm-rubygem-smart_proxy_acd_core
----
. Restart the {SmartProxyServer} services:
+
[source,bash]
----
# systemctl restart foreman-proxy smart_proxy_dynflow_core
----

> . Refresh {SmartProxy} features on {ProjectServer}:

Reviewer: Please work with the installer team to get it integrated in the installer. The `tfm-` instructions are incorrect on EL8 which doesn't have SCLs. `smart_proxy_dynflow_core` is already gone in nightly so this is incorrect. Refreshing of features (including checks that Foreman recognizes the feature) is also part of it. The installer takes care of all of that.

Author: As of right now, I only know how it works on EL7 with Foreman 2.3. 

I will get back to you on how I want to procede soon.

Author: Update: We have plans to integrate ACD into the Foreman installer in the future.

Would you be happy with either a bigger disclaimer (regarding EL7 and Foreman 2.3) or only merging it into the 2.3 branch?

Else, I can only wait until we add it to the Foreman installer and/or support Foreman ACD on a stable Foreman version without `smart_proxy_dynflow_core`, which [seems to be 2.5](https://community.theforeman.org/t/phasing-out-smart-proxy-dynflow-core/22841/8?u=maximilian).

Reviewer: > Update: We have plans to integrate ACD into the Foreman installer in the future.

The patches have been merged to puppet-foreman and puppet-foreman_proxy but there is still a patch missing in foreman-installer. That should really be done.

> Would you be happy with either a bigger disclaimer (regarding EL7 and Foreman 2.3) or only merging it into the 2.3 branch?

I think it's ok to merge a slightly different version for 2.3 but remember that Foreman 2.3 is EOL.

Reviewer 3: @ekohl if you say there need to be a change in foreman-installer, you mean in https://github.com/theforeman/foreman-installer/blob/eb73fb7f4b2fe6470816a8707bae433e2b2cd71d/config/foreman-answers.yaml ?

Reviewer: That and also katello-answers.yaml. If there's a smart proxy component (which there is) then also foreman-proxy-content. Don't forget to add a migration. There's bin/create-migration for that (https://github.com/theforeman/foreman-installer/pull/699 may be good to use as a bugfix).
